### PR TITLE
src/batched: Add AlphaTag compile time selection

### PR DESCRIPTION
**Pull request by e10harvey**
@vqd8a, here are the overload changes.

Related to https://github.com/kokkos/kokkos-kernels/pull/1239.

### CODE REVIEW

**Comment by e10harvey** [comment]
@vqd8a: See 2c65998. Please feel free to close or merge this.

File: src/batched/dense/KokkosBatched_Gemm_Decl.hpp [hunk 1]
```diff
@@ -225,6 +225,12 @@ class BatchedSerialGemm;
 ///                             by tile sizes.
 ///                             BoundsCheck::Yes The functor will     perform bound checks (recommended)
 ///                             BoundsCheck::No  The functor will NOT perform bound checks
+/// \tparam ArgAlphaFmaTag      Specifies whether to apply alpha during fmas.
+///                             AlphaFmaTag::Yes alpha will be applied during fma (C = C * alpha + AB).
+///                             AlphaFmaTag::No  alpha will not be applied during fma (C = C + AB).
+/// \tparam ArgAlphaMulTag      Specifies whether to perform global memory access
+///                             AlphaFmaTag::Yes alpha will be applied during mul (A * B * alpha).
+///                             AlphaFmaTag::No  alpha will not be applied during mul (A * B).
```
Comment (vqd8a) [inline src/batched/dense/KokkosBatched_Gemm_Decl.hpp]: @e10harvey Why do we need two tags ```ArgAlphaFmaTag``` and ```ArgAlphaMulTag```? If not applying alpha during fmas, then it is applying alpha during mul. Can we just use one tag? Do I miss any logic here?

File: src/batched/dense/KokkosBatched_Gemm_Decl.hpp [hunk 2]
```diff
@@ -469,22 +475,50 @@ int BatchedGemm(BatchedGemmHandleType *const handle, const ScalarType alpha,
...
+        constexpr size_t alpha_in_fma_thresh = 64;
+#endif  // __CUDAACC_RDC__
+
+        if (c_m % 32 == 0) {                 // No bounds checking
+          if (c_m >= alpha_in_fma_thresh) {  // apply alpha in fma
+            ret = Impl::BatchedDblBufGemm<ArgTransA, ArgTransB, ArgBatchSzDim,
+                                          BatchedGemmHandleType, ScalarType,
+                                          AViewType, BViewType, CViewType,
+                                          BoundsCheck::No, AlphaTag::Yes,
+                                          AlphaTag::No, tile_m, tile_n, tile_k>(
```
Comment (vqd8a) [inline src/batched/dense/KokkosBatched_Gemm_Decl.hpp]: It is related to my comment above. It is confusing when ```AlphaTag::Yes``` and ```AlphaTag::No``` both appear together.

File: src/batched/dense/impl/KokkosBatched_Gemm_DblBuf_Impl.hpp [hunk 3]
```diff
@@ -241,311 +241,64 @@ class BatchedDblBufGemm {
...
-                        reg_c[m][n] += reg_a[m] * reg_b[n] * __alpha;
-                    }
-                  }
-                });
-          });
+    void __mul(const int m, const int n, view_value_type reg_a[REG_M],
+               view_value_type reg_b[REG_N],
+               view_value_type reg_c[REG_M][REG_N],
+               const AlphaTag::No &) const {
+      reg_c[m][n] += reg_a[m] * reg_b[n];
```
Comment (vqd8a) [inline src/batched/dense/impl/KokkosBatched_Gemm_DblBuf_Impl.hpp]: Is it possible to pass scalars instead of passing arrays?

**Review by vqd8a — COMMENTED** [review]
@e10harvey Thanks for the changes. I just have some minor comments.

File: src/batched/dense/KokkosBatched_Gemm_Decl.hpp [hunk 4]
```diff
@@ -227,10 +227,7 @@ class BatchedSerialGemm;
 ///                             BoundsCheck::No  The functor will NOT perform bound checks
 /// \tparam ArgAlphaFmaTag      Specifies whether to apply alpha during fmas.
 ///                             AlphaFmaTag::Yes alpha will be applied during fma (C = C * alpha + AB).
-///                             AlphaFmaTag::No  alpha will not be applied during fma (C = C + AB).
-/// \tparam ArgAlphaMulTag      Specifies whether to perform global memory access
-///                             AlphaFmaTag::Yes alpha will be applied during mul (A * B * alpha).
-///                             AlphaFmaTag::No  alpha will not be applied during mul (A * B).
+///                             AlphaFmaTag::No  alpha will not be applied during mul (A * B * alpha).
```
Comment (vqd8a) [inline src/batched/dense/KokkosBatched_Gemm_Decl.hpp]: It should be "alpha will be applied during mul"

Comment (e10harvey) [reply src/batched/dense/KokkosBatched_Gemm_Decl.hpp]: Done.

Comment (vqd8a) [reply src/batched/dense/KokkosBatched_Gemm_Decl.hpp]: Thanks, Evan

**Review by vqd8a — APPROVED** [review]

**Pull request merged**